### PR TITLE
MGMT-4210 Refactor ignition code

### PR DESCRIPTION
**Pull request by masayag**
- Extracts discovery image ignition and second day ignition from `bminventory` package to `ignition` package (`IgnitionBuilder`)
- Remove dependency on 'static network config' from bminventory since its functionality is now provided by `IgnitionBuilder`

Signed-off-by: Moti Asayag <masayag@redhat.com>

### CODE REVIEW

**Comment by omertuc** [comment]
Can you please separate it into distinct commits? one for moving the code to another file, and another for making actual changes? It's impossible to know what changed in the way it's currently laid out

**Comment by masayag** [comment]
> Can you please separate it into distinct commits? one for moving the code to another file, and another for making actual changes? It's impossible to know what changed in the way it's currently laid out

I don't see how that can be separated, the entire purpose of this PR is to extract code which was entangled with inventory.go.
In order to decouple it, it should be extracted and changed not to depend on any bminventory in anyway.

I'll update the PR message with the exact changes.

File: internal/ignition/builder.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+	config              IgnitionConfig
+	imageCreateParams   *models.ImageCreateParams
+	logger              logrus.FieldLogger
+	authType            auth.AuthType
+	safeForLogs         bool
+}
+
+// NewBuilder returns ignition builder which builds the ignition file based on the paramters provided
+// to it as functional options during build
+func NewBuilder(staticNetworkConfig staticnetworkconfig.StaticNetworkConfig) *ignitionBuilder {
```
Comment (yevgeny-shnaidman) [inline internal/ignition/builder.go:233]: isn't it a little bit of overkill? Why not just define an API that receives 7 parameters?

Comment (masayag) [reply internal/ignition/builder.go:233]: Having 7 parameters seems already too much.
By using this pattern we benefit of the following:
- No need to change the function signature (unlike with a single function) when new ignition parameter will be needed, just to provide another option to the Build method by the client.
- No need to remember the parameters order 
- Each option function can perform its own validation on the provided value (requires changing the signature to return error). The alternative is a long block of validation for each method parameter
- SafeForLog is 'false' by default, therefore only a single client of the builder will have to provide it as an option to set its value to 'true' (meaning the API can be called with a varied number of arguments)

**Pull request merged**